auth/kerberos/src/krb.cpp

```diff
@@ -348,11 +187,15 @@ std::pair<int, std::string> get_gmsa_krb_ticket( std::string domain_name,
 
     std::pair<size_t, void*> password_found_result =
         Util::find_password( ldap_search_result.second );
+    OPENSSL_cleanse( (void*)ldap_search_result.second.c_str(),
+                     strlen( ldap_search_result.second.c_str() ) );
 
     if ( password_found_result.first == 0 || password_found_result.second == nullptr )
     {
-        std::cerr << Util::getCurrentTime() << '\t' << "ERROR: Password not found" << std::endl;
-        return std::make_pair( -1, std::string( "" ) );
+        std::string log_str = Util::getCurrentTime() + '\t' + "ERROR: Password not found";
+        std::cerr << log_str << std::endl;
+        cf_logger.logger( LOG_ERR, log_str.c_str() );
+        return std::make_pair( -1, log_str );
```

Comment:
Do we want execute_ldapsearch() to return log_str. Seems like earlier, it was just returning an empty string.

Reply:
The return value is a tuple with integer return value (0 is good) and a string describing what could have gone wrong if the return value (-1) shows failure. The caller is able to know log the string if there is a failure.